flake.nix

```
@@ -0,0 +1,44 @@
+{
+description = "rust dev environment";
+
+  outputs = { self, nixpkgs, ... }:
```

```suggestion
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
  };
  
  outputs = { self, nixpkgs, ... }:
```
Let's define the nixpkgs input explicitly.